openshift/machine-config-operator
# Bug 1872001: Stop using EtcdDiscoveryDomain in templates

Etcd is no longer using EtcdDiscoveryDomain and as a result the parameter should be removed. However, the baremetal and friends templates reference it to get the cluster domain. This information is also available from the DNS structure, which is not currently wired into the templates.

This is a rebased replica of #2025. 

**- What I did**
This change adds the DNS struct to the resources in the controller config and switches the templates to get the cluster domain from there.
**- How to verify it**
Install a new cluster without etcdDiscoveryDomain (https://github.com/openshift/installer/pull/4067). It should successfully install the cluster.
**- Description for the changelog**
